src/python/pants/jvm/resolve/coursier_fetch.py

```diff
@@ -566,18 +566,26 @@ class CoursierLockfileForTargetRequest:
 
 
 @rule
-async def get_coursier_lockfile_for_target(
-    request: CoursierLockfileForTargetRequest,
+async def get_coursier_lockfile_for_resolve(
+    coursier_resolve: CoursierResolveKey,
 ) -> CoursierResolvedLockfile:
 
-    coursier_resolve = await Get(CoursierResolve, Targets, request.targets)
-
     lockfile_digest_contents = await Get(DigestContents, Digest, coursier_resolve.digest)
     lockfile_contents = lockfile_digest_contents[0].content
 
     return CoursierResolvedLockfile.from_json_dict(json.loads(lockfile_contents))
 
 
+@rule
+async def get_coursier_lockfile_for_target(
+    request: CoursierLockfileForTargetRequest,
+) -> CoursierResolvedLockfile:
+
+    coursier_resolve = await Get(CoursierResolveKey, Targets, request.targets)
+    lockfile = await Get(CoursierResolvedLockfile, CoursierResolveKey, coursier_resolve)
```

Comment:
Also given there is a rule to go from `CoursierResolveKey` -> `CoursierResolvedLockfile`, could this just be `return await Get(CoursierResolvedLockfile, Targets, request.targets)`?

Reply:
Yeah, assigning to a variable is pointless here :)